Lib/tarfile.py

```diff
@@ -2569,6 +2570,13 @@ def chmod(self, tarinfo, targetpath):
         try:
             os.chmod(targetpath, tarinfo.mode)
         except OSError as e:
+            if hasattr(errno, "EFTYPE") and e.errno == errno.EFTYPE:
+                # On FreeBSD, chmod fails when trying to set the sticky bit on
+                # a file as a normal user. It's a noop in most other platforms.
+                try:
+                    os.chmod(targetpath, tarinfo.mode & ~stat.S_ISVTX)
```

Comment:
if chmod() succeed, you will raise ExtractError. Is it on purpose?

Maybe put your code code inside the try block, so you don't have to copy/paste the except/raise block?